.github/workflows/ci.yml

```diff
@@ -0,0 +1,125 @@
+name: CI
+
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+
+jobs:
+
+  build:
+
+    name: ${{ matrix.name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - name: Meta tests (lint, pr-size, docs)
+            os: ubuntu-latest
+            pyversion: '3.8'
+            TEST_STYLE: 1
+          # Main tests for Python versions
+          - name: Linux py36
+            os: ubuntu-latest
+            pyversion: '3.6'
+            TEST_UNIT: 1
+          - name: Linux py37 full
+            os: ubuntu-latest
+            pyversion: '3.7'
+            TEST_UNIT: 1
+            TEST_FULL: 1
+          - name: Linux py38
+            os: ubuntu-latest
+            pyversion: '3.8'
+            TEST_UNIT_NO_INTERNET: 1
+          - name: Linux py39
+            os: ubuntu-latest
+            pyversion: '3.9'
+            TEST_UNIT: 1
+          # Special OS's and interpreters
+          - name: Linux pypy3
+            os: ubuntu-latest
+            pyversion: 'pypy3'
+            TEST_UNIT: 1
+          - name: MacOS py38
+            os: macos-latest
+            pyversion: '3.8'
+            TEST_UNIT: 1
+          - name: Windows py38
+            os: windows-latest
+            pyversion: '3.8'
+            TEST_UNIT: 1
+
+    steps:
+
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.pyversion }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.pyversion }}
+
+      - name: Install dependencies for style build
+        if: matrix.TEST_STYLE == 1
+        run: |
+            python -m pip install -q -U pip
+            pip install -q -U invoke
+            pip install -q -U flake8 black
+            pip install -q -U sphinx numpydoc
+            pip install -q -U numpy pillow
+
+      - name: Install dependencies for tests
+        if: matrix.TEST_UNIT == 1 || matrix.TEST_UNIT_NO_INTERNET == 1
+        shell: bash
+        run: |
+            python -m pip install -q -U pip
+            pip install -q -U invoke
+            pip install -q -U pytest pytest-cov psutil
+            # Install imagio-ffmpeg because its optional but we still want to test it
+            pip install -q -U imageio-ffmpeg
+            # The first local install is really to get the dependencies (as specified in setup.py)
+            pip install .
+            # This step is to force the download of standard images
+            python setup.py sdist
+            # Install again (with images)
+            pip install .
+            # Delete the local copy, so we're actually using the installed version
+            rm -rf ./imageio ./build ./egg-info
```

Comment:
Makes sense to use the `[extra]` option, but in that case I would do that in the other step where we install optional dependencies (which only runs for one build), because it can potentially slow.

The imageio-ffmpeg is a bit of an exception. We can either change the test so that it's skipped, or manually add it here. I was lazy and opted for the latter for now.

It took me quite some time to get it to working with the standard images. It could well be that we can run the download-script thing here, which would be nicer than the current solution. But for the moment I'm just happy that this works :D

Reply:
Ok, I refactored this bit using `pip install .[dev]` etc.